Meta/Lagom/Tools/CodeGenerators/GMLCompiler/PropertyDefinition.cpp

```diff
@@ -0,0 +1,336 @@
+        case SimpleType::U64:
+            if (!property.is_integer<u64>())
+                return PropertyError::format("Non-integer value {} supplied for u64 property", property);
+            return TRY(String::formatted("static_cast<u64>({})", property.as_integer<u64>()));
+        case SimpleType::Double:
+            if (!property.is_number())
+                return PropertyError::format("Non-number value {} supplied for double property", property);
+            return TRY(property.as_number().visit(
+                // NOTE: Passing by mutable reference here in order to disallow implicit casts.
+                [](u64& value) { return String::formatted("static_cast<u64>({})", value); },
+                [](i64& value) { return String::formatted("static_cast<i64>({})", value); },
+                [](double& value) { return String::formatted("static_cast<double>({})", value); }));
+        case SimpleType::String:
+            if (!property.is_string())
+                return PropertyError::format("Non-string value {} supplied for string property", property);
+            return TRY(String::formatted(R"~~~("{}"_string)~~~", TRY(escape_string(property))));
+        case SimpleType::ByteString:
+            if (!property.is_string())
+                return PropertyError::format("Non-string value {} supplied for byte string property", property);
+            return TRY(String::formatted(R"~~~("{}"sv)~~~", TRY(escape_string(property))));
+        case SimpleType::Bitmap:
+            if (!property.is_string())
+                return PropertyError::format("Non-string value {} supplied for bitmap property", property);
+            return TRY(String::formatted(R"~~~(TRY(Gfx::Bitmap::load_from_file("{}"sv)))~~~", TRY(escape_string(property))));
+        case SimpleType::Color: {
+            if (!property.is_string())
+                return PropertyError::format("Non-string value {} supplied for color property", property);
+            auto maybe_color = Gfx::Color::from_string(property.as_string());
+            if (!maybe_color.has_value())
+                return PropertyError::format("Invalid color {}", property);
+            auto const color = maybe_color.release_value();
+            return TRY(String::formatted("::Gfx::Color {{ {}, {}, {}, {} }}", color.red(), color.green(), color.blue(), color.alpha()));
+        }
+        case SimpleType::UIDimension: {
+            auto maybe_dimension =  GUI::UIDimension::construct_from_json_value(property);
+            if (!maybe_dimension.has_value())
+                return PropertyError::format("Invalid UI dimension {}", property);
+            return TRY(maybe_dimension->as_cpp_source());
+        }
+        case SimpleType::Margins: {
+            if (!property.is_array())
+                return PropertyError::format("Non-array value {} supplied for margins property {}", property, name());
+            auto const element_count = property.as_array().size();
```

Comment:
How about checking that `element_count >= 1 && element_count <= 4` instead of relying on compilation error?